tephra-core/src/test/java/co/cask/tephra/TransactionAdminTest.java

```diff
@@ -0,0 +1,195 @@
+
+  @BeforeClass
+  public static void start() throws Exception {
+    zkServer = InMemoryZKServer.builder().setDataDir(tmpFolder.newFolder()).build();
+    zkServer.startAndWait();
+
+    conf = HBaseConfiguration.create();
+    conf.setBoolean(TxConstants.Manager.CFG_DO_PERSIST, false);
+    conf.set(TxConstants.Service.CFG_DATA_TX_ZOOKEEPER_QUORUM, zkServer.getConnectionStr());
+    conf.set(TxConstants.Service.CFG_DATA_TX_CLIENT_RETRY_STRATEGY, "n-times");
+    conf.setInt(TxConstants.Service.CFG_DATA_TX_CLIENT_ATTEMPTS, 1);
+
+    Injector injector = Guice.createInjector(
+      new ConfigModule(conf),
+      new ZKModule(),
+      new DiscoveryModules().getDistributedModules(),
+      Modules.override(new TransactionModules().getDistributedModules())
+        .with(new AbstractModule() {
+          @Override
+          protected void configure() {
+            bind(TransactionStateStorage.class).to(InMemoryTransactionStateStorage.class).in(Scopes.SINGLETON);
+          }
+        }),
+      new TransactionClientModule()
+    );
+
+    zkClientService = injector.getInstance(ZKClientService.class);
+    zkClientService.startAndWait();
+
+    // start a tx server
+    txService = injector.getInstance(TransactionService.class);
+    txClient = injector.getInstance(TransactionSystemClient.class);
+    try {
+      LOG.info("Starting transaction service");
+      txService.startAndWait();
+    } catch (Exception e) {
+      LOG.error("Failed to start service: ", e);
+    }
+  }
+
+  @Before
+  public void reset() throws Exception {
+    txClient.resetState();
+  }
+
+  @AfterClass
+  public static void stop() throws Exception {
+    txService.stopAndWait();
+    zkClientService.stopAndWait();
+    zkServer.stopAndWait();
+  }
+
+  @Test
+  public void testPrintUsage() throws Exception {
+    ByteArrayOutputStream out = new ByteArrayOutputStream();
+    ByteArrayOutputStream err = new ByteArrayOutputStream();
+    TransactionAdmin txAdmin = new TransactionAdmin(new PrintStream(out), new PrintStream(err));
+    txAdmin.doMain(new String[0], conf);
+    //noinspection ConstantConditions
+    Assert.assertTrue(Bytes.toString(err.toByteArray()).startsWith("Usage:"));
+    Assert.assertEquals(0, out.toByteArray().length);
+  }
+
+  @Test
+  public void testTruncateInvalidTx() throws Exception {
+    Transaction tx1 = txClient.startLong();
+    Transaction tx2 = txClient.startShort();
+    txClient.invalidate(tx1.getWritePointer());
+    txClient.invalidate(tx2.getWritePointer());
+    
+    ByteArrayOutputStream out = new ByteArrayOutputStream();
+    ByteArrayOutputStream err = new ByteArrayOutputStream();
+    TransactionAdmin txAdmin = new TransactionAdmin(new PrintStream(out), new PrintStream(err));
+    txAdmin.doMain(new String[]{"--truncate-invalid-tx", String.valueOf(tx2.getWritePointer())}, conf);
+    List<String> expected = ImmutableList.of("Invalid list size before truncation: 2",
```

Comment:
I'm asserting on the output to see what exactly gets shown to the user when the command is run. As I have seen cases before when the output message or its format changes inadvertently, and no one notices it. Is there a better way to do this?